### PR TITLE
[FFI/Test_JDK22] Add test suites intended for union

### PR DESCRIPTION
**Pull request by ChengJin01**
The changes add all the same test suites intended
for union in JDK21 to JDK22 with the updated APIs
introduced in JDK22.

Signed-off-by: ChengJin01 <jincheng@ca.ibm.com>

### CODE REVIEW

**Comment by ChengJin01** [comment]
The changes (which are the same as the test suites for union at https://github.com/eclipse-openj9/openj9/pull/18291)
are verified in personal builds (JDK21 & 22) on all supported platforms.

**Comment by ChengJin01** [comment]
Reviewer: @tajila
FYI: @keithc-ca, @pshipton

File: test/functional/Java22andUp/src/org/openj9/test/jep454/downcall/UnionStructTests.java [hunk 1]
```diff
@@ -0,0 +1,946 @@
...
+			double result = (double)mh.invokeExact(3336.333D, unionSegmt);
+			Assert.assertEquals(result, 14483.218D, 0.001D);
+		}
+	}
+
+	@Test
+	public void test_add2DoubleStructs_returnStruct_Nested2DoubleUnion() throws Throwable {
+		GroupLayout structLayout = MemoryLayout.structLayout(JAVA_DOUBLE.withName("elem1"), JAVA_DOUBLE.withName("elem2"));
+		VarHandle doubleHandle1 = structLayout.varHandle(PathElement.groupElement("elem1"));
+		VarHandle doubleHandle2 = structLayout.varHandle(PathElement.groupElement("elem2"));
```
Comment (keithc-ca) [inline test/functional/Java22andUp/src/org/openj9/test/jep454/downcall/UnionStructTests.java:829]: Please remove these unused locals and the import of `VarHandle`, and do the same for 1Java21Only`.

File: test/functional/Java22andUp/src/org/openj9/test/jep454/upcall/UpcallMethodHandles.java [hunk 2]
```diff
@@ -542,8 +915,7 @@ public static MemorySegment createNewCharFromCharAndCharFromPtr_RetPtr(MemorySeg
 		int diff = (charArg2 >= charArg1) ? (charArg2 - charArg1) : (charArg1 - charArg2);
 		diff = (diff > 5) ? 5 : diff;
 		char result = (char)(diff + 'A');
-		MemorySegment resultSegmt = arena.allocate(JAVA_CHAR);
-		resultSegmt.set(JAVA_CHAR, 0, result);
+		MemorySegment resultSegmt = arena.allocateFrom(JAVA_CHAR, (char)result);
```
Comment (keithc-ca) [inline test/functional/Java22andUp/src/org/openj9/test/jep454/upcall/UpcallMethodHandles.java:918]: Please remove the redundant cast to `char`. Lines 948, 976 and 1075 also have redundant casts.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: test/functional/Java22andUp/src/org/openj9/test/jep454/downcall/UnionStructTests.java (hunk 1 above)
Comment (ChengJin01) [reply test/functional/Java22andUp/src/org/openj9/test/jep454/downcall/UnionStructTests.java]: Fixed.

**Comment by keithc-ca** [comment]
Jenkins test sanity alinux jdk21

**Comment by keithc-ca** [comment]
Jenkins test sanity aix,alinux,plinux,xlinux,zlinux,win jdk21,jdknext

**Review by tajila — APPROVED** [review]

**Comment by ChengJin01** [comment]
[1] https://openj9-jenkins.osuosl.org/job/Test_openjdknext_j9_sanity.functional_ppc64_aix_Personal/12/artifact/Test_openjdknext_j9_sanity.functional_ppc64_aix_Personal_testList_1/12/aqa-tests/TKG/output_16989554223828/Test_openjdknext_j9_sanity.functional_ppc64_aix_Personal_testList_1.tap/*view*/
```
ok 44 - Jep454Tests_testLinkerFfi_DownCall_0
  ---
    duration_ms: 9176
  ...
ok 45 - Jep454Tests_testLinkerFfi_UpCall_0
  ---
    duration_ms: 5770
```
[2] https://openj9-jenkins.osuosl.org/job/Test_openjdknext_j9_sanity.functional_aarch64_linux_Personal/25/tapTestReport/
```
51 - - Jep454Tests_testLinkerFfi_DownCall_0 	8.2 sec	OK
60 - - Jep454Tests_testLinkerFfi_UpCall_0 	12 sec	OK

All Failed Tests
Test Name	Duration	Age
 227 - - Jep425Tests_testVirtualThread_1 	9.5 sec	
 228 - - Jep425Tests_testVirtualThread_2 	9.3 sec	
 71 - - cmdLineTester_criu_nonPortableRestore_4 	6 min 58 sec	
 138 - - Jep425Tests_testVirtualThread_0 	7.5 sec	
 139 - - Jep425Tests_testVirtualThread_3 	7.5 sec	
```
which means the failing tests in the builds above are irrelevant to our test suites.

**Comment by ChengJin01** [comment]
[1] https://openj9-jenkins.osuosl.org/job/Test_openjdk21_j9_sanity.functional_s390x_linux_Personal/50/tapTestReport/
```
42 - - Jep442Tests_testLinkerFfi_DownCall_0 	23 sec	OK	
49 - - Jep442Tests_testLinkerFfi_UpCall_0 	3.8 sec	OK
```
[2] https://openj9-jenkins.osuosl.org/job/Test_openjdknext_j9_sanity.functional_s390x_linux_Personal/41/tapTestReport/
```
15 - - Jep454Tests_testLinkerFfi_DownCall_0 	11 sec	OK	
230 - - Jep454Tests_testLinkerFfi_UpCall_0 	12 sec	OK	
```

**Review by keithc-ca — APPROVED** [review]
This only changes tests and all those tests passed.

**Pull request merged**